src/test/java/com/wikia/webdriver/common/core/api/ArticleContent.java

```diff
@@ -1,47 +1,53 @@
 package com.wikia.webdriver.common.core.api;
 
-import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
 import java.util.ArrayList;
 
-import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.ResponseHandler;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
 import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.openqa.selenium.WebDriverException;
 
 import com.wikia.webdriver.common.contentpatterns.PageContent;
+import com.wikia.webdriver.common.core.Helios;
 import com.wikia.webdriver.common.core.TestContext;
-import com.wikia.webdriver.common.core.XMLReader;
 import com.wikia.webdriver.common.core.configuration.Configuration;
 import com.wikia.webdriver.common.core.helpers.User;
 import com.wikia.webdriver.common.core.url.UrlBuilder;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+
+import java.net.URL;
+import java.util.Iterator;
 
 public class ArticleContent extends ApiCall {
 
-  private static String secret;
+  private static String setupURL = new UrlBuilder().getUrlForWiki(Configuration.getWikiName())
+      + "/api.php?action=query&prop=info&format=json&intoken=edit&titles=";
   private String baseURL = new UrlBuilder().getUrlForWiki(Configuration.getWikiName())
-      + "/wikia.php?controller=Wikia\\Helios\\SampleController&method=edit&title=";
+      + "/api.php?action=edit&title=";
   private ArrayList<BasicNameValuePair> params = new ArrayList<>();
   private User user = User.STAFF;
+  private String editToken;
+  private static String EDIT_TOKEN_ERROR_MESSAGE = "Problem with edit token API call";
 
   public ArticleContent() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1186)